src/db.ts

```diff
@@ -0,0 +1,354 @@
+            REFERENCES ${CONVERSATIONS}(conversation_id)
+ );
+`;
+
+const CREATE_SESSIONS_TABLE_QUERY = `
+CREATE TABLE IF NOT EXISTS ${SESSIONS}(
+    session_id         VARCHAR(36) NOT NULL,
+    participant_id     VARCHAR(36) NOT NULL,
+    media_type         VARCHAR(20) NOT NULL,
+    direction          VARCHAR(8) NOT NULL,
+    peer_id            VARCHAR(36),
+    provider           VARCHAR(255) NOT NULL,
+    requested_routings VARCHAR(1000),
+    used_routing       VARCHAR(255),
+    selected_agent_id  VARCHAR(36),
+    remote             VARCHAR(255),
+    PRIMARY KEY(session_id),
+    CONSTRAINT participant_id_fk
+        FOREIGN KEY(participant_id) 
+            REFERENCES ${PARTICIPANTS}(participant_id)
+ );
+`;
+
+// segment_id is session_id + _ + segment_start + _ + segment_end
+const CREATE_SEGMENTS_TABLE_QUERY = `
+CREATE TABLE IF NOT EXISTS ${SEGMENTS}(
+    segment_id      VARCHAR(150) NOT NULL,
+    session_id      VARCHAR(36) NOT NULL,
+    segment_start   VARCHAR(25) NOT NULL,
+    segment_end     VARCHAR(25) NOT NULL,
+    segment_type    VARCHAR(20) NOT NULL,
+    conference      BOOLEAN NOT NULL,
+    disconnect_type VARCHAR(20),
+    queue_id        VARCHAR(36),
+    PRIMARY KEY(segment_id),
+    CONSTRAINT session_id_fk
+        FOREIGN KEY(session_id) 
+            REFERENCES ${SESSIONS}(session_id)
+);
+`;
+// metric_id is session_id + _ + name + _ + value
+const CREATE_METRICS_TABLE_QUERY = `
+CREATE TABLE IF NOT EXISTS ${METRICS}(
+    metric_id  VARCHAR(150) NOT NULL,
+    session_id VARCHAR(36) NOT NULL,
+    name       VARCHAR(20) NOT NULL,
+    value      INT NOT NULL,
+    emit_date  VARCHAR(25) NOT NULL,
+    PRIMARY KEY(metric_id),
+    CONSTRAINT session_id_fk
+        FOREIGN KEY(session_id) 
+            REFERENCES ${SESSIONS}(session_id)
+);
+`;
+
+// Insertion Queries
+const INSERT_CONVERSATIONS_TABLE_QUERY = `
+INSERT INTO ${CONVERSATIONS}(conversation_id, conversation_start, conversation_end, originating_direction, division_ids) VALUES($1, $2, $3, $4, $5) RETURNING *
+`;
+
+const INSERT_PARTICIPANTS_TABLE_QUERY = `
+INSERT INTO ${PARTICIPANTS}(participant_id, conversation_id, participant_name, purpose, user_id) VALUES($1, $2, $3, $4, $5) RETURNING *
+`;
+
+const INSERT_SESSIONS_TABLE_QUERY = `
+INSERT INTO ${SESSIONS}(session_id, participant_id, media_type, direction, peer_id, provider, requested_routings, used_routing, selected_agent_id, remote) VALUES($1, $2, $3, $4, $5, $6, $7, $8, $9, $10) RETURNING *
+`;
+
+const INSERT_SEGMENTS_TABLE_QUERY = `
+INSERT INTO ${SEGMENTS}(segment_id, session_id, segment_start, segment_end, segment_type, conference, disconnect_type, queue_id) VALUES($1, $2, $3, $4, $5, $6, $7, $8) RETURNING *
+`;
+
+const INSERT_METRICS_TABLE_QUERY = `
+INSERT INTO ${METRICS}(metric_id, session_id, name, value, emit_date) VALUES($1, $2, $3, $4, $5) RETURNING *
+`;
+
+// Fetch queries
+const FETCH_CONVERSATION_QUERY = `
+SELECT * FROM ${CONVERSATIONS}
```

Comment:
It might be nice to create a view for this. A view will allow you to run `SELECT * FROM conversations_view WHERE conversation_id=$1` to the same effect as this query. The view would be defined as the query here without the WHERE clause.